File: Wikipedia/Code/WMFWelcomeFadeInAndUpOnceViewController.m
#import "WMFWelcomeFadeInAndUpOnceViewController.h"
#import "UIView+WMFWelcomeFadeInAndUp.h"
#import "Wikipedia-Swift.h"

@interface WMFWelcomeFadeInAndUpOnceViewController ()

@property (strong, nonatomic) IBOutlet UIView *containerView;
@property (strong, nonatomic) NSNumber *fadeInAndUpDelay;

@end

@implementation WMFWelcomeFadeInAndUpOnceViewController

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    if (!self.hasAlreadyFaded) {
        [self.containerView wmf_zeroLayerOpacity];
    }
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    if (!self.hasAlreadyFaded) {
        [self.containerView wmf_fadeInAndUpAfterDelay:[self.fadeInAndUpDelay floatValue]];
    }
    self.hasAlreadyFaded = YES;
}

- (IBAction)next:(id)sender {
    [self.delegate showNextWelcomePage:self];
}

@end

File: Wikipedia/Code/WMFWelcomeFadeInAndUpOnceViewController.swift

class WMFWelcomeFadeInAndUpOnceViewController: UIViewController {

    var hasAlreadyFaded = false
    weak var welcomeNavigationDelegate:WMFWelcomeNavigationDelegate? = nil
    
    @IBOutlet private var fadeInAndUpDelay:NSNumber! = 0
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        if (!hasAlreadyFaded) {
            view.wmf_zeroLayerOpacity()
        }
    }

    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        
        if (!hasAlreadyFaded) {
            view.wmf_fadeInAndUpAfterDelay(CGFloat(fadeInAndUpDelay))
        }
        hasAlreadyFaded = true
    }
    
    @IBAction func next(withSender sender: AnyObject) {
        welcomeNavigationDelegate?.showNextWelcomePage(self)
    }
}
